src/plugins/discourse/fetch.js

```diff
@@ -303,6 +363,50 @@ function failForNotOk(response: Response) {
   }
 }
 
+/**
+ * Parses a "latest" topic.
+ *
+ * A "latest" topic, is a topic as returned by the /latest.json API call,
+ * and has two distinct assumptions:
+ * - bumped_at is always present.
+ * - the posters field uses IDs instead of usernames.
+ *
+ * usernamesById map used to resolve these IDs to usernames.
+ */
+function parseLatestTopic(
+  usernamesById: Map<number, string>,
+  json: any
+): Topic {
+  // TODO: this is a code smell.
+  // Currently we're using a quirk of the API code, that the first poster in the summary
+  // is always the original poster. The only other way to tell is by parsing it from the
+  // translated human description.
+  // See: https://github.com/discourse/discourse/blob/23367e79ea735598766ec6f507f6132e0bad3dba/lib/topic_query.rb#L443
+  const opUsername = usernamesById.get(json.posters[0].user_id);
+  if (opUsername == null) {
+    throw new Error(
```

Comment:
If it happens, it would be a Discourse bug. As there should always be a first element in the array. And every user ID in the result set should always be present in the mapping to usernames. Which is why I've added this error.

My manual tests have been:
- https://sourcecred-test.discourse.group
- https://discourse.sourcecred.io
- https://discourse.sustainoss.org

As they're loadable in minutes when I need a cache clear. And covers the deep-than-wide approach of prioritizing our own forums.

I agree, once we've got all these components in place it would be good to test how robust it is on substantially larger forums such as:
- https://forum.zcashcommunity.com
- https://users.rust-lang.org

For now I feel like requiring a several hour stress test is not in line with deep-then-wide and would be better done as an iteration later on or as bug reports come in.

Reply:
Note: should we choose to remove this field, because of the flimsy detection assumptions (other comment), this check would also be removed.